src/main/java/tr/com/srdc/cda2fhir/jolt/ResourceAccumulator.java

```diff
@@ -36,25 +36,90 @@ public Object transform(Object input, Map<String, Object> context) {
 			return null;
 		}
 		Map<String, Object> resource = (Map<String, Object>) input;
+
 		if (resource == null) {
 			resource = new LinkedHashMap<String, Object>();
 		}
+
+		List<Object> resources = (List<Object>) context.get("Resources");
+		if (resources == null) {
+			resources = new ArrayList<Object>();
+			context.put("Resources", resources);
+		}
+
+		int id = resources.size() + 1;
+
+		String reference = String.format("%s/%s", resourceType, id);
+		String display = AdditionalModifier.getDisplay(resource);
+
+		Map<String, Object> resourceMap = (Map<String, Object>) context.get("RESOURCE_MAP");
+
+		if (resourceMap == null) {
+			resourceMap = new HashMap<String, Object>();
+			context.put("RESOURCE_MAP", resourceMap);
+		}
+
+		List<Object> identifiers;
+		if (resource.get("identifier") instanceof LinkedHashMap) {
+			identifiers = new ArrayList<Object>();
+			identifiers.add(resource.get("identifier"));
+		} else {
+			identifiers = (List<Object>) resource.get("identifier");
+
+		}
+
 		if (input != null && "Medication".equals(resourceType)) {
-			CodeableConceptMap medicationMap = (CodeableConceptMap) context.get("MEDICATION_MAP");
-			if (medicationMap != null) {
-				Map<String, Object> existing = medicationMap.get(resource.get("code"));
-				if (existing != null) {
-					return existing;
+			MedicationMap medicationMap = (MedicationMap) context.get("MEDICATION_MAP");
+			Map<String, Map<String, Object>> referenceOrgMap = (Map<String, Map<String, Object>>) context
+					.get("REF_ORG_MAP");
+
+			if (medicationMap == null) {
+				medicationMap = new MedicationMap();
+				context.put("MEDICATION_MAP", medicationMap);
+			}
+
+			if (referenceOrgMap == null) {
+				referenceOrgMap = new HashMap<String, Map<String, Object>>();
+				context.put("REF_ORG_MAP", referenceOrgMap);
+			}
+
+			Map<String, String> manufacturer = (Map<String, String>) resource.get("manufacturer");
+			List<Object> orgIdentifiers = null;
+
+			if (manufacturer != null) {
+				String manuRef = manufacturer.get("reference");
+				Map<String, Object> organization = referenceOrgMap.get(manuRef);
+				if (organization != null) {
+					orgIdentifiers = (List<Object>) resource.get("identifier");
 				}
 			}
+			Map<String, Object> existing = medicationMap.get(resource, orgIdentifiers);
+			if (existing != null) {
```

Comment:
I do not claim to understand what you are doing and the problems you faced but this is a very odd way of doing things. Why would you relate organization deduplication to Medication deduplication? If we are deduplicating meds based on code only and organizations are different there is no way to put the manufacturer reference right to medication so either look at the organization together with the  code to deduplicate or just bite the bullet and have one of the medications have the wrong reference and do organization deduplication separately.

Reply:
We had a conversation about this, but for posterity, the organization dedup is maintained in the CDAIIResourceMap and the medication dedup is maintained separately in the CDACDResourceMap. The IIMap used inside of the MedicationContainer class is used to dedup medications with Organizations so that the same medications with different manufacturers are treated as unique medication instances. But this map is not used to deduplicate organizations as a whole.